pandas/core/arrays/timedeltas.py

```diff
@@ -162,15 +162,32 @@ def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
         result._freq = freq
         return result
 
-    def __new__(cls, values, freq=None, dtype=_TD_DTYPE):
+    def __new__(cls, values, freq=None, dtype=_TD_DTYPE, copy=False):
 
         freq, freq_infer = dtl.maybe_infer_freq(freq)
 
-        values = np.array(values, copy=False)
-        if values.dtype == np.object_:
-            values = array_to_timedelta64(values)
+        values, inferred_freq = sequence_to_td64ns(
+            values, copy=copy, unit=None)
+        if inferred_freq is not None:
+            if freq is not None and freq != inferred_freq:
+                raise ValueError('Inferred frequency {inferred} from passed '
+                                 'values does not conform to passed frequency '
+                                 '{passed}'
+                                 .format(inferred=inferred_freq,
+                                         passed=freq.freqstr))
+            elif freq is None:
+                # TODO: should this be the stronger condition `if freq_infer`?
+                #  i.e what if the user passed `freq=None` and specifically
+                #  wanted freq=None in the result?
```

Comment:
I am not sure this is a valid condition.  what are you trying to do here?

Reply:
I doubt a user setting a freq is actually used anywhere, not even sure of the purpose of it; nor is it likely tested.

This is just adding complextiy.

Reply:
OK, I'll remove the comment